pinot-core/src/main/java/org/apache/pinot/core/operator/docvalsets/ProjectionBlockValSet.java

```diff
@@ -50,6 +53,26 @@ public ProjectionBlockValSet(DataBlockCache dataBlockCache, String column, DataS
     _dataBlockCache = dataBlockCache;
     _column = column;
     _dataSource = dataSource;
+    NullValueVectorReader nullValueReader = _dataSource == null ? null : _dataSource.getNullValueVector();
```

Comment:
Data source should never be `null`. The computation should be lazy, and the operator should decide whether to read it based on if null handling is enabled

Reply:
Updated to the computation lazily. 

Please note that passing whether query has null handling enabled is tedious (need to pass QueryContext through many classes). Let me know if you still prefer to pass it through.

Reply:
Moved the logic to `getNullBitmap()` method. Please note that this method is called only when `nullHandlingEnabled` is set to true.